.github/workflows/applications.yml

```diff
@@ -0,0 +1,282 @@
+
+name: Build and publish application binaries
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+    - '**'
+    - '!.github/**'
+    - '.github/workflows/applications.yml'
+
+  pull_request_target:
+    types: [ opened ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["9.8.2", "9.10.1", '9.6.6']
+        cabal: ["3.12"]
+        os: ["ubuntu-20.04", "ubuntu-22.04", "macos-14"]
+        cabalcache: ["true"]
+
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.kadena_cabal_cache_aws_access_key_id }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.kadena_cabal_cache_aws_secret_access_key }}
+      AWS_DEFAULT_REGION: us-east-1
+
+      # Aritfacts
+      ARTIFACT_BUCKET: kadena-cabal-cache
+      BINFILE: pact.${{ matrix.ghc }}.${{ matrix.os }}.${{ github.sha }}.tar.gz
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    # Non Haskell dependencies
+
+    # Install Z3
+    - name: Install z3 (ubuntu-20.04)
```

Comment:
Not needed

Reply:
ok; are we still even building for 20.04 or should we be doing 22.04 and 24.04?

Reply:
@emilypi

Reply:
doing 22.04 now